conan/tools/_compilers.py

```diff
@@ -396,3 +403,36 @@ def _cppstd_mcst_lcc(mcst_lcc_version, cppstd):
             "17": v17, "gnu17": vgnu17,
             "20": v20, "gnu20": vgnu20}.get(cppstd)
     return "-std=%s" % flag if flag else None
+
+
+def _cppstd_intel_cc(_, cppstd):
+    """
+    Inspired in:
+    https://software.intel.com/content/www/us/en/develop/documentation/
+    oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/
+    compiler-options/compiler-option-details/language-options/std-qstd.html
+    """
+    # Note: for now, we don't care about compiler version
+    v98 = "c++98"
+    vgnu98 = "gnu++98"
+    v03 = "c++03"
+    vgnu03 = "gnu++03"
+    v11 = "c++11"
+    vgnu11 = "gnu++11"
+    v14 = "c++14"
+    vgnu14 = "gnu++14"
+    v17 = "c++17"
+    vgnu17 = "gnu++17"
+    v20 = "c++20"
+    vgnu20 = "gnu++20"
+    v23 = "c++2b"
+    vgnu23 = "gnu++2b"
+
+    flag = {"98": v98, "gnu98": vgnu98,
+            "03": v03, "gnu03": vgnu03,
+            "11": v11, "gnu11": vgnu11,
+            "14": v14, "gnu14": vgnu14,
+            "17": v17, "gnu17": vgnu17,
+            "20": v20, "gnu20": vgnu20,
+            "23": v23, "gnu23": vgnu23}.get(cppstd, None)
+    return "-std=%s" % flag if flag else None
```

Comment:
For now and as far as I know, there are no limitations to use any of these flags for the new Intel oneAPI C++/DPC++/Classis compilers